driver/gl/draw.go

```diff
@@ -9,66 +9,52 @@ import (
 	"github.com/go-gl/gl/v3.2-core/gl"
 )
 
-// walkObjects will walk an object tree executing the passed function on all objects encountered.
-// The childrenFirst bool dictates whether to call f() on the obj parameter first,
-// or obj's children (if any) first
-func (c *glCanvas) walkObjects(obj fyne.CanvasObject, pos fyne.Position, childrenFirst bool,
-	f func(object fyne.CanvasObject, pos fyne.Position)) {
-
+// walkObjects will walk an object tree executing the passed functions following the following rules:
+// - fBeforeChildren is called for the start obj before traversing its children
+// - the obj's children are traversed by calling walkObjects on each of them
+// - fAfterChildren is called for the obj after traversing the obj's children
+// The walk can be aborted by returning true in one of the functions:
+// - if fBeforeChildren returns true, further traversing is stopped immediatly, the after function
+//   will not be called for the obj where the walk stopped, however, it will be called for all its
+//   parents
+// - if a walk has been stopped, the after function is called with the third argument set to true
+func (c *glCanvas) walkObjects(
+	obj fyne.CanvasObject,
+	pos fyne.Position,
+	fBeforeChildren func(fyne.CanvasObject, fyne.Position) bool,
```

Comment:
We don't use type prefix for variables, so perhaps beforeChildren or callbackBefore may be more in-keeping?